rolocController.cpp

```diff
@@ -197,50 +224,62 @@ void ROLOCcontroller::processRolocData()
 #endif
     qint16 rolocData = rolocGetData();
 
-    if(mCurrentMode == ROLOC::eMODE_GET_SIGNAL_STRENGTH)
+    if (rolocData != BAD_DATA_READ)
     {
-        // update the sig strength
-        mROLOCsignalStrenth = rolocData;
-
-        //qDebug() << "sig strength = " << mROLOCsignalStrenth; // TODO verify if is this working ?
-
-        // clear the depth measurements
-        mDepthAccumulator.clear();
-    }
-    else
-    {
-        mDepthAccumulator.append(rolocData);
-        mNumSamples++;
-
-        if(mNumSamples > N_MULTI_LF_DEPTH_SAMPLE)
+        if(mCurrentMode == ROLOC::eMODE_GET_SIGNAL_STRENGTH)
         {
-            QList<quint8> acceptedDepthReadings;
-            float scaleOfElimination = (float)N_MULTI_LF_DEPTH_DELTA;
-
-            double mean = getMean(mDepthAccumulator);
-            double stdDev = qSqrt(getVariance(mDepthAccumulator));
-
-            qDebug() << "Mean: " << mean;
-            qDebug() << "Standard Dev.: " << stdDev;
+            // update the sig strength
+            mROLOCsignalStrenth = rolocData;
+#if DBG_BLOCK
+            qDebug() << "sig strength = " << mROLOCsignalStrenth; // TODO verify if is this working ?
+#endif
+            // clear the depth measurements
+            mDepthAccumulator.clear();
+            sendDataReport();
+        }
+        else
+        if (mCurrentMode == ROLOC::eMODE_GET_DEPTH_MEASUREMENT)
+        {
+            mDepthAccumulator.append(rolocData);
+            mNumSamples++;
 
-            for(quint8 i = 0; i < mDepthAccumulator.count(); i++)
+            if(mNumSamples > N_MULTI_LF_DEPTH_SAMPLE)
             {
-                quint8 isLessThanLowerBound = mDepthAccumulator.at(i) < mean - stdDev * scaleOfElimination;
-                quint8 isGreaterThanUpperBound = mDepthAccumulator.at(i) > mean + stdDev * scaleOfElimination;
-                quint8 isOutOfBounds = isLessThanLowerBound || isGreaterThanUpperBound;
+                QList<quint8> acceptedDepthReadings;
+                float scaleOfElimination = (float)N_MULTI_LF_DEPTH_DELTA;
 
-                if(!isOutOfBounds)
+                double mean = getMean(mDepthAccumulator);
+                double stdDev = qSqrt(getVariance(mDepthAccumulator));
+#if DBG_BLOCK
+                qDebug() << "*************************Distance Stats*****************";
+                qDebug() << "*************************** bit set  -> " <<  CHECK_BIT(rolocData, 1)  << "*********************";
+                qDebug() << "smaple size: " << mNumSamples;
+                qDebug() << "sample size: " << mNumSamples;
+                qDebug() << "Mean: " << mean;
+                qDebug() << "Standard Dev.: " << stdDev;
```

Comment:
you have a duplicate debug here. Lets remove one
```suggestion
                qDebug() << "sample size: " << mNumSamples;
                qDebug() << "Mean: " << mean;
                qDebug() << "Standard Dev.: " << stdDev;
```